Source/WebCore/layout/formattingContexts/inline/InlineLineTypes.h

```diff
@@ -33,6 +33,12 @@ namespace Layout {
 class InlineItem;
 
 using InlineBoxBoundaryTextSpacings = WTF::HashMap<size_t, float, DefaultHash<size_t>, WTF::UnsignedWithZeroKeyHashTraits<size_t>>;
+using TrimmableTextSpacings = InlineBoxBoundaryTextSpacings;
+
+struct TextSpacingContext {
+    InlineBoxBoundaryTextSpacings inlineBoxBoundaryTextSpacings;
+    TrimmableTextSpacings trimmableTextSpacings;
```

Comment:
I think the main question/concern I have is in regards to the relationship between these two fields within a `TextSpacingContext`.  

Are trimmable text spacings supposed to be a subset of the inline box boundary text spacings? If so, I would imagine it would be a bit tricky to make sure that these two are always in a valid state at any given point in time as it seems like these two things can get mutated arbitrarily.  Should we maybe consider changing what an “InlineBoxBoundaryTextSpacing" is and have something on it that indicates whether it is trimmable or not?

Reply:
These are used independently. One is for identifying spacing between box boundaries, the other is to identify trimmable spacing between text runs.

`InlineBoxBoundaryTextSpacings` maps inline item indexes to spacing. It serves to identify start boxes that mark spacing between boxes boundaries. If we have start box with an entry here, this means that we have to add spacing between this start box and the previous box. We need to account for it during layout and to reproduce such spacing among 2 boxes during display. Therefore, this serve for identifying spacing between box boundaries such that we can both: add these spacing into layout/display and trim this spacing during line breaking if needed.

`TrimmableTextSpacings` map inline item indexes to spacing. Text content can be split into multiple different runs text items. This structure serves to identify spacing that was added to a text item because of its adjacency to another text item. For example, imagine we have adjacent text items A and B, and the last character of A and the first character of B require spacing to be added between them. This spacing is always added as a leading spacing to B. However, during line breaking, if B gets placed on a different line as A, they are no longer adjacent and we must trim such spacing. We don't use this map for actually adding the spacing, just for trimming it. Therefore this maps serves only to identify trimmable spacing between text runs. Maybe this can be called `TrimmableTextSpacingBetweenTextItems` or something.